nixos/modules/config/fonts/fontdir.nix

```diff
@@ -21,27 +25,42 @@ in
 {
 
   options = {
+    fonts.fontDir = {
 
-    fonts = {
-
-      enableFontDir = mkOption {
+      enable = mkOption {
         type = types.bool;
         default = false;
         description = ''
           Whether to create a directory with links to all fonts in
-          <filename>/run/current-system/sw/share/X11-fonts</filename>.
+          <filename>/run/current-system/sw/share/X11/fonts</filename>.
         '';
       };
 
-    };
+      decompressFonts = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to decompress fonts in
+          <filename>/run/current-system/sw/share/X11/fonts</filename>.
+        '';
+      };
 
+    };
   };
 
-  config = mkIf config.fonts.enableFontDir {
+  config = mkIf cfg.enable {
 
+    # This is enough to make a symlink because the xserver
+    # module already links all /share/X11 paths.
     environment.systemPackages = [ x11Fonts ];
 
-    environment.pathsToLink = [ "/share/X11-fonts" ];
+    services.xserver.filesSection = ''
+      FontPath "${x11Fonts}/share/X11/fonts"
+    '';
+
+    programs.xwayland.defaultFontPath = "${x11Fonts}/share/X11/fonts";
```

Comment:
If xwayland is enabled setting `fonts.fontsDir.enable` will implicitly cause a rebuild and a user could be pretty clueless: I would move this to the xwayland module. It may be reasonable to set the default value of `programs.xwayland.defaultFontPath` to
```
optionalString config.fonts.fontDir.enable "/run/current-system/sw/share/X11/fonts"
```

Reply:
done